### DISCUSSION TITLE
how to config plugins

### DISCUSSION THREAD

**Discussion by wilmerxu** [dotansimha/graphql-code-generator#6288] [Q&A] [answered]
import { codegen } from '@graphql-codegen/core';
const config = {
    documents: [
      {
        location: 'operation.graphql',
        document: parse(documents),
      },
    ],
    config: {
      withHooks: true,
    },
    filename: outputFile,
    schema: parse(printSchema(schema)),
    plugins: [
      {
        'typescript': {},
        'typescript-react-apollo': {},
        'typescript-operations': {},
      },
    ],
    pluginMap: {
      'typescript': typescriptPlugin,
      'typescript-operations': typescriptOperationsPlugin,
      'typescript-react-apollo': typescriptReactApolloPlugin,
    },
  };

**but typescript-operations and typescript-react-apollo can not take effect**

**Comment by ardatan** [comment, marked as answer]
Please share a minimal and clear reproduction on CodeSandbox.
See https://github.com/dotansimha/graphql-code-generator/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by wilmerxu** [comment]
this problem have already solved:

```
const typescript = require('@graphql-codegen/typescript');
const typescriptOperations = require('@graphql-codegen/typescript-operations');
const typescriptReactApollo = require('@graphql-codegen/typescript-react-apollo');

.....

plugins: [
      { typescript: {} },
      { typescriptOperations: {} },
      { typescriptReactApollo: {} }
    ],
    pluginMap: {
      typescript,
      typescriptOperations,
      typescriptReactApollo,
    },
```
